### PR TITLE
Carl/donet2

### PR DESCRIPTION
**Pull request by CarlVerret**
There's something I don't understand with Log2SoftwareFallback as it was pushed previously.  I pushed a workaround for now.  All test are passing.

### CODE REVIEW

**Comment by CarlVerret** [comment]
git workflow will soon be modified for .net 4.7.1 support

File: TestcsFastFloat/BaseTestClass.cs [hunk 1]
```diff
@@ -1,9 +1,7 @@
 ﻿using ApprovalTests;
 using ApprovalTests.Namers;
 using ApprovalTests.Reporters;
-using Newtonsoft.Json;
```
Comment (lemire) [inline TestcsFastFloat/BaseTestClass.cs:4]: Yes, let us clean the includes as much as possible.

File: TestcsFastFloat/Basic/TestDoubleParser.cs [hunk 2]
```diff
@@ -180,7 +188,7 @@ public void cas_compute_float_64_1()
         if (p == 23)
           p++;
 
-     //   var sut = new DoubleBinaryFormat();
+        //   var sut = new DoubleBinaryFormat();
```
Comment (lemire) [inline TestcsFastFloat/Basic/TestDoubleParser.cs:191]: Before the next release, let us clean all these commented out lines.

File: csFastFloat/Utils/Utils.cs [hunk 3]
```diff
@@ -213,5 +217,64 @@ internal unsafe static bool strncasecmp(byte* input1, ReadOnlySpan<byte> input2,
...
+      {
+        return 32 + Log2SoftwareFallback((uint)value);
+      }
+ 
+      return Log2SoftwareFallback(hi);
+      
+      [MethodImpl(MethodImplOptions.AggressiveInlining)]
+      static int Log2SoftwareFallback(uint value)
+      {
+       if (value == 0)         return 32;
```
Comment (lemire) [inline csFastFloat/Utils/Utils.cs:238]: Before the next release, the code should get reformatted, here it appears that the indentation is changes from line to line.

File: csFastFloat/Utils/Utils.cs [hunk 4]
```diff
@@ -213,5 +217,64 @@ internal unsafe static bool strncasecmp(byte* input1, ReadOnlySpan<byte> input2,
...
+      int n = 1;
+      if (value >> 16 == 0) { n += 16; value <<= 16; }
+      if (value >> 24 == 0) { n +=  8; value <<=  8; }
+      if (value >> 28 == 0) { n +=  4; value <<=  4; }
+      if (value >> 30 == 0) { n +=  2; value <<=  2; }
+      n -= (int) (value >> 31);
+      return n;
```
Comment (lemire) [inline csFastFloat/Utils/Utils.cs:245]: I did not validate this algorithm (I am going to trust you). It seems reasonable. It is not going to be fast, but that's fine.

**Comment by lemire** [comment]
@CarlVerret Why did you remove the -0 tests? We do want to parse -0 to -0. The platform can do what it likes, but we should check that we do it properly, like .NET 5.

See [How does your programming language handle “minus zero” (-0.0)?](https://lemire.me/blog/2021/03/04/how-does-your-programming-language-handle-minus-zero-0-0/)

**Comment by CarlVerret** [comment]
Actually I moved it to another test thats doesn't depends on displayed value

**Comment by lemire** [comment]
> Actually I moved it to another test thats doesn't depends on displayed value 

Good!

File: csFastFloat/Utils/Utils.cs (hunk 4 above)
Comment (CarlVerret) [reply csFastFloat/Utils/Utils.cs:245]: I would prefer to understand why the previous version wasn't working as it should...  at least we've got a workaround for now.

Comment (lemire) [reply csFastFloat/Utils/Utils.cs]: I would not waste a lot of time on it. If you are convinced that it is something important, just go with what works, and create an issue explaining your concerns. Someone might pick it up. If not, your solution is fine. I would absolutely release 2.0 with this new code of yours.

**Pull request merged**